File: backend/app/schemas/member.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict

from app.models.enums import MemberRole


class TripMemberBase(BaseModel):
    user_id: Optional[str] = None  # UUID string
    name: str
    origin_city: str
    role: MemberRole = MemberRole.member


class TripMemberCreate(TripMemberBase):
    pass


class TripMemberOut(TripMemberBase):
    model_config = ConfigDict(from_attributes=True)
    id: str
    trip_id: str
    created_at: datetime

File: frontend/src/types/member.ts
export type MemberRole = "member" | "admin";

export interface TripMemberCreateRequest {
  user_id?: string | null;
  name: string;
  origin_city: string;
  role?: MemberRole;
}

export interface TripMember {
  id: string;
  trip_id: string;
  user_id: string | null;
  name: string;
  origin_city: string;
  role: MemberRole;
  created_at: string;
}

export interface TripTransportUpdateRequest {
  price: number;
}

export interface TripTransport {
  id: string;
  trip_id: string;
  member_id: string;
  price: number;
  created_at: string;
}
